Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -385,20 +423,48 @@ private Order GetOrderByIdInternal(int orderId)
         public Order GetOrderByBrokerageId(string brokerageId)
         {
             // this function can be invoked by brokerages when getting open orders, guard against null ref
-            if (_orders == null) return null;
+            if (_closedOrders == null || _openOrders == null) return null;
 
-            var order = _orders.FirstOrDefault(x => x.Value.BrokerId.Contains(brokerageId)).Value;
+            var order =
+                _openOrders.FirstOrDefault(x => x.Value.BrokerId.Contains(brokerageId)).Value ??
+                _closedOrders.FirstOrDefault(x => x.Value.BrokerId.Contains(brokerageId)).Value;
             return order != null ? order.Clone() : null;
         }
 
         /// <summary>
         /// Gets all orders matching the specified filter
         /// </summary>
         /// <param name="filter">Delegate used to filter the orders</param>
-        /// <returns>All open orders this order provider currently holds</returns>
+        /// <returns>All orders this order provider currently holds</returns>
```

Comment:
The existing comment wasn't correct either, but technically it returns `All orders this order provider current holds filtered by the specified filter` -- should update this here, the interface, and other implementations that have the same comment

Reply:
Will do 👍